nikola/plugins/task/galleries.py

```diff
@@ -393,6 +409,62 @@ def create_galleries(self):
                 'uptodate': [utils.config_changed(self.kw.copy(), 'nikola.plugins.task.galleries:mkdir')],
             }
 
+    def find_order(self, gallery):
+        """
+        If there is an IDX file for the gallery, use that to determine
+        the order in which images shall be displayed in the gallery.
+        Returns a ordered list or None.
+        """
+        idx_path = os.path.join(gallery, "IDX")
+        order = []
+        if os.path.isfile(idx_path):
+            with open(idx_path, "r") as idx_file:
+                for line in idx_file.readlines():
+                    order.append(line.strip())
+        else:
+            order = None
+        return order
+
+    def find_captions(self, gallery):
+        """
+        Return a list of image url and a dict of {image url: caption} if
+        we have a captions file in the gallery folder. A captions file
+        ("captions") contains
+
+        one url, whitespace, text for the caption
+
+        with one entry for EACH file in the gallery. Image files without
+        captions should use ::empty::  as the caption text. The order of
+        images in the captions file is assumed to be the order in which
+        the user wishes the images displayed on the page.
+       
+        Example:
+        
+        header.jpg This is the header image for my blog
```

Comment:
This format won’t cut it. Filenames could contain spaces.  Instead of a custom parser, use something like JSON or YAML.